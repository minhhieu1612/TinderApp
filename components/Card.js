import React, {useState} from 'react';
import {View, Image, Text, StyleSheet} from 'react-native';
import {Transitioning, Transition} from 'react-native-reanimated';
import {COLORS} from '../constants/styles';
import {Icon} from 'react-native-elements';

const ANIMATION_DURATION = 200;

const transition = (
  <Transition.Sequence>
    <Transition.Out
      type="slide-bottom"
      durationMs={ANIMATION_DURATION}
      interpolation="easeIn"
    />
    <Transition.Together>
      <Transition.In
        type="fade"
        durationMs={ANIMATION_DURATION}
        delayMs={ANIMATION_DURATION / 2}
      />
      <Transition.In
        type="slide-bottom"
        durationMs={ANIMATION_DURATION}
        delayMs={ANIMATION_DURATION / 2}
        interpolation="easeOut"
      />
    </Transition.Together>
  </Transition.Sequence>
);

const Card = ({card, transitionRef}) => {
  // console.log(card.picture.replace(/^http:\/\//i, 'https://'));
  const [active, setActive] = useState(0);
  const {name, location, picture, phone, registered, gender} = card;
  const arrDetail = [
    {
      key: 1,
      title: 'My name is',
      content: `${name.title} ${name.last} ${name.first}`,
      icon: {name: 'face'},
    },
    {
      key: 2,
      title: 'My gender is',
      content: `${gender}`,
      icon: {name: 'event-note'},
    },
    {
      key: 3,
      title: 'My address is',
      content: `${location.street}`,
      icon: {name: 'location-on'},
    },
    {
      key: 4,
      title: 'My phone number is',
      content: `${phone}`,
      icon: {name: 'phone'},
    },
    {
      key: 5,
      title: 'My registered key is',
      content: `${registered}`,
      icon: {name: 'lock'},
    },
  ];
  return (
    <View style={styles.card}>
      <View style={styles.cardHeader} />
      <View style={styles.wrapImage}>
        <Image
          source={{uri: picture.replace(/^http:\/\//i, 'https://')}}
          style={styles.cardImage}
        />
      </View>
      <View style={styles.cardBody}>
        <Transitioning.View
          ref={transitionRef}
          transition={transition}
          style={styles.bottomContainerMeta}>
          <View style={styles.detail}>
            <Text style={styles.detailTitle}>{arrDetail[active].title}</Text>
            <Text style={styles.detailContent}>
              {arrDetail[active].content}
            </Text>
          </View>
        </Transitioning.View>
        <View style={styles.listIcon}>
          <View style={styles.elementTop(active)} />
          {arrDetail.map(({icon, key}, index) => (
            <Icon
              key={key}
              type="material"
              name={icon.name}
              color={active === index ? COLORS.PRIMARY : COLORS.DARK(2)}
              onPress={() => {
                transitionRef.current.animateNextTransition();
                setActive(index);
              }}
              size={35}
            />
          ))}
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    position: 'relative',
    flex: 0.8,
    backgroundColor: '#fff',
    borderRadius: 5,
    shadowRadius: 30,
    shadowOpacity: 0.6,
    shadowColor: '#000',
    shadowOffset: {width: 200, height: 20},
    alignItems: 'center',
  },
  cardHeader: {
    flex: 0.35,
    borderTopLeftRadius: 5,
    borderTopRightRadius: 5,
    backgroundColor: 'rgba(0,0,0,.15)',
    width: 'auto',
    alignSelf: 'stretch',
    borderBottomWidth: 1,
    borderBottomColor: 'rgba(0,0,0,.3)',
  },
  wrapImage: {
    position: 'absolute',
    width: 170,
    height: 170,
    borderRadius: 100,
    top: 40,
    flex: 0.3,
    backgroundColor: 'white',
    padding: 5,
    borderWidth: 1,
    borderColor: 'rgba(0,0,0,.3)',
  },
  cardImage: {
    flex: 1,
    borderRadius: 100,
    resizeMode: 'cover',
  },
  cardBody: {
    flex: 0.65,
    width: 'auto',
    borderBottomLeftRadius: 5,
    borderBottomRightRadius: 5,
    backgroundColor: 'transparent',
    alignSelf: 'stretch',
    padding: 20,
    alignItems: 'center',
  },
  bottomContainerMeta: {
    flex: 0.6,
    alignSelf: 'stretch',
    marginBottom: 10,
  },
  detail: {
    flex: 1,
    alignSelf: 'stretch',
    alignItems: 'center',
    justifyContent: 'flex-end',
    // marginBottom: 10,
  },
  detailTitle: {
    fontSize: 18,
    color: 'gray',
    fontWeight: '500',
  },
  detailContent: {
    fontSize: 26,
    fontWeight: '700',
    textAlign: 'center',
  },
  listIcon: {
    flex: 0.4,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
    alignSelf: 'stretch',
  },
  elementTop: (index) => ({
    position: 'absolute',
    top: 15,
    left: 56 * index,
    width: 55,
    height: 7,
    flex: 1,
    alignSelf: 'stretch',
    backgroundColor: COLORS.PRIMARY,
  }),
});

export default Card;
