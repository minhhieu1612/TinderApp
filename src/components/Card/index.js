import React, {useState} from 'react';
import {View, Image, Text, StyleSheet} from 'react-native';
import {COLORS} from '../../constants/styles';
import {Icon} from 'react-native-elements';
import getValue from '../../helpers/getValue';

const Card = ({card}) => {
  const [active, setActive] = useState(0);
  const name = getValue(card, 'name'),
    location = getValue(card, 'location'),
    picture = getValue(card, 'picture'),
    phone = getValue(card, 'phone'),
    registered = getValue(card, 'registered'),
    gender = getValue(card, 'gender');
  const fullName = name ? `${name.title} ${name.last} ${name.first}` : name;
  const address = location ? location.street : '';
  const arrDetail = [
    {
      key: 1,
      title: 'My name is',
      content: fullName,
      icon: {name: 'face'},
    },
    {
      key: 2,
      title: 'My gender is',
      content: `${gender}`,
      icon: {name: 'event-note'},
    },
    {
      key: 3,
      title: 'My address is',
      content: address,
      icon: {name: 'location-on'},
    },
    {
      key: 4,
      title: 'My phone number is',
      content: `${phone}`,
      icon: {name: 'phone'},
    },
    {
      key: 5,
      title: 'My registered key is',
      content: `${registered}`,
      icon: {name: 'lock'},
    },
  ];
  const imgSource = picture
    ? {uri: picture.replace(/^http:\/\//i, 'https://')}
    : require('./placeholder.jpg');
  return (
    <View style={styles.card}>
      <View style={styles.cardHeader} />
      <View style={styles.wrapImage}>
        <Image source={imgSource} style={styles.cardImage} />
      </View>
      <View style={styles.cardBody}>
        <View style={styles.bottomContainerMeta}>
          <View style={styles.detail}>
            <Text style={styles.detailTitle}>{arrDetail[active].title}</Text>
            <Text style={styles.detailContent}>
              {arrDetail[active].content}
            </Text>
          </View>
        </View>
        <View style={styles.listIcon}>
          <View style={styles.elementTop(active)} />
          {arrDetail.map(({icon, key}, index) => (
            <Icon
              key={key}
              type="material"
              name={icon.name}
              color={active === index ? COLORS.PRIMARY : COLORS.DARK(2)}
              onPress={() => {
                setActive(index);
              }}
              size={35}
            />
          ))}
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    position: 'relative',
    flex: 0.8,
    backgroundColor: '#fff',
    borderRadius: 5,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 3,
    },
    shadowOpacity: 0.29,
    shadowRadius: 4.65,
    elevation: 5,
    alignItems: 'center',
    zIndex: 0,
  },
  cardHeader: {
    flex: 0.35,
    borderTopLeftRadius: 5,
    borderTopRightRadius: 5,
    backgroundColor: 'rgba(0,0,0,.15)',
    width: 'auto',
    alignSelf: 'stretch',
    borderBottomWidth: 1,
    borderBottomColor: 'rgba(0,0,0,.3)',
  },
  wrapImage: {
    position: 'absolute',
    width: 170,
    height: 170,
    borderRadius: 100,
    top: 40,
    flex: 0.3,
    backgroundColor: 'white',
    padding: 5,
    borderWidth: 1,
    borderColor: 'rgba(0,0,0,.3)',
  },
  cardImage: {
    flex: 1,
    backgroundColor: COLORS.DARK(4),
    width: 158,
    height: 158,
    alignSelf: 'stretch',
    borderRadius: 100,
    resizeMode: 'cover',
  },
  cardBody: {
    flex: 0.65,
    width: 'auto',
    borderBottomLeftRadius: 5,
    borderBottomRightRadius: 5,
    backgroundColor: 'transparent',
    alignSelf: 'stretch',
    padding: 20,
    alignItems: 'center',
  },
  bottomContainerMeta: {
    flex: 0.6,
    alignSelf: 'stretch',
    marginBottom: 10,
  },
  detail: {
    flex: 1,
    alignSelf: 'stretch',
    alignItems: 'center',
    justifyContent: 'flex-end',
  },
  detailTitle: {
    fontSize: 18,
    color: 'gray',
    fontWeight: '500',
  },
  detailContent: {
    fontSize: 26,
    fontWeight: '700',
    textAlign: 'center',
  },
  listIcon: {
    flex: 0.4,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
    alignSelf: 'stretch',
  },
  elementTop: (index) => ({
    position: 'absolute',
    top: 15,
    left: 56 * index,
    width: 55,
    height: 7,
    flex: 1,
    alignSelf: 'stretch',
    backgroundColor: COLORS.PRIMARY,
  }),
});

export default Card;
