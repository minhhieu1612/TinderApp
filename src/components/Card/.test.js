import React from 'react';
import renderer from 'react-test-renderer';
import Card from './index';

it('test Component Card', () => {
  const card = [
    {
      gender: 'female',
      name: {
        title: 'miss',
        first: 'catherine',
        last: 'kelley',
      },
      location: {
        street: '1352 ninth st',
        city: 'springfield',
        state: 'iowa',
        zip: '96474',
      },
      email: 'catherine.kelley56@example.com',
      username: 'greenrabbit328',
      password: 'susie',
      salt: 'CSr3dKeE',
      md5: 'ec90a7ad03fbd883416460ebdf59bca1',
      sha1: 'c083947d97432acf0f9e107e2b5ca1c134fbed0a',
      sha256:
        'a4d5938460490956082af6182b7ecbb88bf3610a6e7b1f54649e2220976ff055',
      registered: '1409024352',
      dob: '458529607',
      phone: '(724)-575-1036',
      cell: '(511)-192-7949',
      SSN: '578-55-1922',
      picture: 'http://api.randomuser.me/portraits/women/43.jpg',
    },
  ];
  expect(renderer.create(<Card card={card} />)).toMatchSnapshot();
});
