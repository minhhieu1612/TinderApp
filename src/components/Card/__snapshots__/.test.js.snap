// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Component Card 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 5,
      "elevation": 5,
      "flex": 0.8,
      "position": "relative",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.29,
      "shadowRadius": 4.65,
      "zIndex": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "rgba(0,0,0,.15)",
        "borderBottomColor": "rgba(0,0,0,.3)",
        "borderBottomWidth": 1,
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "flex": 0.35,
        "width": "auto",
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "rgba(0,0,0,.3)",
        "borderRadius": 100,
        "borderWidth": 1,
        "flex": 0.3,
        "height": 170,
        "padding": 5,
        "position": "absolute",
        "top": 40,
        "width": 170,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/components/Card/placeholder.jpg",
        }
      }
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "rgba(0,0,0,0.4)",
          "borderRadius": 100,
          "flex": 1,
          "height": 158,
          "resizeMode": "cover",
          "width": 158,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "transparent",
        "borderBottomLeftRadius": 5,
        "borderBottomRightRadius": 5,
        "flex": 0.65,
        "padding": 20,
        "width": "auto",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 0.6,
          "marginBottom": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flex": 1,
            "justifyContent": "flex-end",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "gray",
              "fontSize": 18,
              "fontWeight": "500",
            }
          }
        >
          My name is
        </Text>
        <Text
          style={
            Object {
              "fontSize": 26,
              "fontWeight": "700",
              "textAlign": "center",
            }
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "flex": 0.4,
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#86b300",
            "flex": 1,
            "height": 7,
            "left": 0,
            "position": "absolute",
            "top": 15,
            "width": 55,
          }
        }
      />
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              brand={false}
              solid={false}
              style={
                Array [
                  Object {
                    "color": "#86b300",
                    "fontSize": 35,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              brand={false}
              solid={false}
              style={
                Array [
                  Object {
                    "color": "rgba(0,0,0,0.2)",
                    "fontSize": 35,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              brand={false}
              solid={false}
              style={
                Array [
                  Object {
                    "color": "rgba(0,0,0,0.2)",
                    "fontSize": 35,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              brand={false}
              solid={false}
              style={
                Array [
                  Object {
                    "color": "rgba(0,0,0,0.2)",
                    "fontSize": 35,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              brand={false}
              solid={false}
              style={
                Array [
                  Object {
                    "color": "rgba(0,0,0,0.2)",
                    "fontSize": 35,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
